MinecraftClient/ChatBots/MineCube.cs

```diff
@@ -0,0 +1,340 @@
+﻿using System;
+using System.Collections.Generic;
+using MinecraftClient.Mapping;
+using MinecraftClient.Inventory;
+using System.Threading;
+
+namespace MinecraftClient.ChatBots
+{
+    class MineCube : ChatBot
+    {
+        public override void Initialize()
+        {
+            if (!GetTerrainEnabled())
+            {
+                LogToConsole(Translations.Get("extra.terrainandmovement_required"));
+                UnloadBot();
+                return;
+            }
+            RegisterChatBotCommand("mine", "Mine a cube from a to b", "/mine x y z OR /mine x1 y1 z1 x2 y2 z2", EvaluateMineCommand);
+            RegisterChatBotCommand("mineup", "Walk over a flat cubic platform of blocks and mine everything above you", "/mine x1 y1 z1 x2 y2 z2 (y1 = y2)", EvaluateMineUpCommand);
+        }
+
+        /// <summary>
+        /// Dig out a 2 Block high cube and let the bot walk through it
+        /// mining all blocks above it that it can reach.
+        /// </summary>
+        /// <param name="walkingArea">Area that the bot should walk through. (The lower Y coordinate of the 2 high cube.)</param>
+        public void MineUp(Cube walkingArea)
+        {
+            Material2Tool m2t = new Material2Tool();
```

Comment:
Please read my feedback about static class below first.

As the class is static, you do not need to create a class instance before using the class.